### DISCUSSION TITLE
company and ESS: suggest variable name even without $

### DISCUSSION THREAD

**Discussion by maxecharel** [company-mode/company-mode#1274] [Q&A] [answered]
Hi, while company suggests variable names with `df$`, 
![company_works](https://user-images.githubusercontent.com/61969138/144023680-a746a1a2-a935-4f10-be46-429e1ad43c21.png)
However, for packages such as `ggplot2` variables names are entered directly. In that case company does not suggest variable names from the related dataframe:
![company_does_not_work](https://user-images.githubusercontent.com/61969138/144023957-05c4f2b9-d64d-48b9-83c8-99bfe1a936e0.png)
(NB: `dist` is not a variable from the df dataframe; I rather expect company to suggest the variable `disp`)

Is this feature included and I haven't enabled it? Or should I submit a feature request?

**Comment by yugaego** [comment]
Hi, what's the output of `M-x company-diag` in the same context/example?

"Used Backend" in it shows who is responsible for the exact completion candidates list, see [Bug Report template](https://github.com/company-mode/company-mode/issues/new?assignees=&labels=&template=bug-report.md&title=) for more details.

**Reply by maxecharel** [reply]
Hi @yugaego and thx for your suggestion.
The output is the following:
```
Emacs 27.2 (x86_64-pc-linux-gnu) of 2021-11-27 on <blanked>
Company 0.9.13

company-backends: ((company-R-library company-R-args company-R-objects :separate)
 company-bbdb company-semantic company-cmake company-clang company-files
 (company-dabbrev-code company-gtags company-etags company-keywords)
 company-oddmuse company-dabbrev)

Used backend: (company-R-library company-R-args company-R-objects :separate)

Major mode: ess-r-mode
Prefix: "dis"
Completions:
  #("discoveries" 0 11 (company-backend company-R-objects))
  #("discrete_scale" 0 14 (company-backend company-R-objects))
  #("dist" 0 4 (company-backend company-R-objects))
```
Should I report this as a bug?

**Reply by dgutov** [reply]
You can try reporting it to `ESS`. where all the backends from the "Used backend:" list come from.

https://github.com/emacs-ess/ESS/blob/master/lisp/ess-r-completion.el#L359-L417

It's probably a bug, but could be by design, for all I know.

**Reply by maxecharel** [reply]
Hi @dgutov, thx for the suggestion. I've reported it as a bug: https://github.com/emacs-ess/ESS/issues/1171

**Comment by yugaego** [comment, marked as answer]
To summarize, the issue was identified on `ESS` side as a duplicate of https://github.com/emacs-ess/ESS/issues/253 .

**Reply by maxecharel** [reply]
@yugaego well summarized, this is actually a feature that has to be implemented in ESS
